lib/java/com/google/android/material/drawable/DrawableUtils.java

```diff
@@ -308,23 +285,40 @@ public static Drawable compositeTwoLayeredDrawable(
       }
     }
 
-    drawable.setLayerSize(1, topLayerNewWidth, topLayerNewHeight);
-    drawable.setLayerGravity(1, Gravity.CENTER);
+    LayerDrawable drawable;
+    if (VERSION.SDK_INT >= VERSION_CODES.M) {
+      drawable = new LayerDrawable(new Drawable[] {bottomLayerDrawable, topLayerDrawable});
+
+      drawable.setLayerSize(1, topLayerNewWidth, topLayerNewHeight);
+      drawable.setLayerGravity(1, Gravity.CENTER);
+    } else {
+      Drawable scaledTopLayerDrawable =
+          new ScaledDrawableWrapper(topLayerDrawable, topLayerNewWidth, topLayerNewHeight);
```

Comment:
This requires API >= 23, see https://developer.android.com/reference/android/graphics/drawable/DrawableWrapper so cannot be used here

Reply:
This code is executed only in API >= 23:

https://github.com/material-components/material-components-android/blob/58f5439f15a6d33ae82a8e5d162f32ffbedea275/lib/java/com/google/android/material/drawable/DrawableUtils.java#L289-L294

Reply:
I'm talking about `ScaledDrawableWrapper` which extends `DrawableWrapper` which is only API >= 23, there seems to be a `DrawableWrapperCompat` though

Reply:
Yes, that was the compat `DrawableWrapper`:

https://github.com/material-components/material-components-android/blob/58f5439f15a6d33ae82a8e5d162f32ffbedea275/lib/java/com/google/android/material/drawable/ScaledDrawableWrapper.java#L5

Anyway, I've rebased the PR and updated this class.